source/github-helpers/get-default-branch.ts

```diff
@@ -23,12 +24,16 @@ const getDefaultBranch = cache.function(async function (repository?: pageDetect.
 
 	if (arguments.length === 0 || JSON.stringify(repository) === JSON.stringify(getRepo())) {
 		if (pageDetect.isRepoHome()) {
-			const currentBranch = select('#branch-select-menu [data-menu-button]');
+			const currentBranch = await elementReady('[data-hotkey="w"]');
 			if (currentBranch) { // If missing, it'll default to the API call
-				return currentBranch.textContent!;
+				return currentBranch.title;
 			}
 		}
 
+		if (pageDetect.utils.getRepositoryInfo()!.path === 'commits') {
```

Comment:
isRepoCommitList?

Reply:
No, we can only ensure we are on the default branch if it is exactly `/owner/repo/commits`. `/owner/repo/commits/branch` is not deterministic.

Reply:
Oh, yeah, specifically we can only know the default branch’s name if we _are_ on the default branch. Let's add this note to the code (inside `getCurrentBranchFromFeed` though)
```suggestion
		// Not `isRepoCommitList` because this works exclusively on the default branch
		if (pageDetect.utils.getRepositoryInfo()!.path === 'commits') {
```